src/Document/Query.php

```diff
@@ -88,7 +88,7 @@ public static function cssToXpath($path)
 
         // Arbitrary attribute value contains whitespace
         $path = preg_replace_callback(
-            '/\[\S+["\'](.+)["\']\]/',
+            '/\[\S+?["\'](.+?)["\']\]/',
```

Comment:
Just thinking about this regexp here, and I think we should use backreference here to match match always `"..."` or `'...'` and not `"...'` or `'..."`. But it's separate issue/improvement ...

Reply:
@weierophinney It is, yes. But that has nothing to with this PR. That problem already existed and should probably be fixed in a separate PR. If you want it to be fixed right here, you're welcome to update this one. I'm not good enough with regular expressions.